### ISSUE TITLE
Custom input layers support for Integrated Gradients

### ISSUE THREAD

**Issue by gipster** [SeldonIO/alibi#287]
At the moment the [`IntegratedGradients`](https://github.com/SeldonIO/alibi/blob/782ad53504d347f92c0ceb1cfbf5d1c376ceb677/alibi/explainers/integrated_gradients.py#L305) class only supports tf.keras and keras models taking numpy arrays as inputs and in which the input layer is a standard keras input layer such as [`tf.keras.Input`](https://www.tensorflow.org/api_docs/python/tf/keras/Input) or [`tf.keras.layers.InputLayer`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/InputLayer). 

Investigation is ongoing to see whether it is possible to extend the method to models with custom input layers taking other types of inputs such as dictionaries.

[Related issue](https://github.com/SeldonIO/alibi/issues/285)

**Comment by ofiryaish** [comment]
Hey, what about input of a model that is list of of numpy arrays and scalars ( in certain point they are I use concatenate merge layers)
